api/src/main/java/io/cloudevents/CloudEvent.java

```diff
@@ -33,45 +33,60 @@
 @ParametersAreNonnullByDefault
 public interface CloudEvent {
 
-    /**
-     * The event context attributes
-     */
-    Attributes getAttributes();
-
-    /**
-     * The event data
-     */
+	/**
+	 *
+	 * @return the event data, effectively representing the payload of this event.
+	 */
     @Nullable
     byte[] getData();
 
     /**
-     * The event extensions
-     * <p>
-     * Extensions values could be String/Number/Boolean
+     * @return the version of the Cloud Events specification which this event uses.
      */
-    Map<String, Object> getExtensions();
+    SpecVersion getSpecVersion();
```

Comment:
After this change, `Attributes` role becomes minor, and I can even see the interface being removed (there is no such "interface" in the spec and the attributes are defined next to other fields like `data` anyways)

Does it make sense to keep `Attributes` in type `CloudEvent` type at all?

Reply:
Personally I tend to agree with Sergey, but would also like to have a separate discussion on that.

Reply:
in go we make the top level event implement the attributes and also an accessor to get the holder of attributes, if that was translated to this lib, it would be that `CloudEvents extends Attributes` and also has a getter `getAttributes() Attributes `

Reply:
@n3wscott it kind of contradicts your previous comment, but it could very well be me misinterpreting this comment for you simply stating how things are currently done in go.